ts/features/wallet/paypal/screen/PaypalDetailScreen.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from "react";
+import { NavigationInjectedProps } from "react-navigation";
+import { connect } from "react-redux";
+import BasePaymentMethodScreen from "../../common/BasePaymentMethodScreen";
+import PaymentMethodFeatures from "../../component/features/PaymentMethodFeatures";
+import PaypalCard from "../PaypalCard";
+import { GlobalState } from "../../../../store/reducers/types";
+import { payPalByIdSelector } from "../../../../store/reducers/wallet/wallets";
+
+type NavigationParams = Readonly<{
+  idWallet: number;
+}>;
+
+type Props = NavigationInjectedProps<NavigationParams> &
+  ReturnType<typeof mapStateToProps>;
+
+/**
+ * Detail screen for a paypal payment method
+ * @constructor
+ */
+const PaypalDetailScreen: React.FunctionComponent<Props> = props => {
+  const paypal = props.paymentMethod(props.navigation.getParam("idWallet"));
+  // it should not never happen since this screen is shown from a navigation that starts from a paypal payment method
+  if (paypal === undefined) {
+    return null;
```

Comment:
```suggestion
    return <WorkunitGenericFailure />;
```
In order to avoid that if an incorrect number is passed, a white screen is displayed. Result:

https://user-images.githubusercontent.com/26501317/142844082-ef0fb6a3-367f-46b6-8736-5e6eaca13416.mov

Furthermore, since the wallet ID is not typed like the other identifiers used, for example, `EUCovidCertificateId`, using a `number` as navigation parameter could lead to errors in future usages. For this reason could be worth considering to use as navigation params the `PayPalPaymentMethod` and using the `walletId` to load the payment method from the store. While the usage is somewhat redundant it would ensure that no wrong `number` are passed as a navigation argument.

Reply:
Good point. Deal!
These suggestions should be addressed with
a5354fb6e06b96ae28b4d574116afaeb58dfcd14
and then
a5354fb6e06b96ae28b4d574116afaeb58dfcd14

Reply:
In 5aa0aac the `PayPalPaymentMethod` it is no longer read by the store. It must be read by the store using the `walletId` in order to correcly update the Payment and favourite section.

Reply:
oh sh*t
I introduced a bug while applying the suggestion
Fixed here 6e268c378072b73505d957b5ff178b40d316ab0c